### PR TITLE
Bug 1905128: Fix prow's e2e test script

### PR DESCRIPTION
**Pull request by irosenzw**
Signed-off-by: Ido Rosenzwig <irosenzw@redhat.com>

### CODE REVIEW

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
Looks like we have another failure:

> Performance test: downloads new bundle for operator-hub
> Expected 141580 to be less than 120000.

cc @TheRealJon

**Comment by spadgett** [comment]
This could be related to #7233.

We might bump the limit or remove the performance tests altogether. I'm not sure how much value they provide.

**Comment by spadgett** [comment]
It looks like the Insights card has broken the dashboard tests since the Skeleton is not going away. See the gray circle by Insights in the screenshot: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_console/7446/pull-ci-openshift-console-master-e2e-gcp-console/1336063939183644672/artifacts/e2e-gcp-console/test/gui_test_screenshots/fb820ad08fc7418496dfbdfcfe0b7a8f.png

Probably introduced in #7351. cc @bond95

**Comment by spadgett** [comment]
The dashboard tests wait for all skeletons to disappear: https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/test-views/dashboard-shared.view.ts#L6-L10

**Comment by spadgett** [comment]
@bond95 FYI, we had a bug in our test script which prevented the tests from running on your PR. We should determine if this isn't a bug though since Insights status never seems to load.

**Comment by spadgett** [comment]
I pushed a commit to temporarily disable the tests so we can get Protractor running again.

**Comment by bond95** [comment]
> @bond95 FYI, we had a bug in our test script which prevented the tests from running on your PR. We should determine if this isn't a bug though since Insights status never seems to load.

@spadgett In dev environment everything is working fine for me, I could check through cluster-bot if Insights working.

**Comment by bond95** [comment]
@spadgett hmmm... seems like in CI cluster version Insights tab not loaded 
image

**Comment by spadgett** [comment]
Insights fix in #7491 (thanks @bond95 !)

**Comment by spadgett** [comment]
/hold

It looks like the ConsoleExternalLogLink test now fails because of an image pull error on the pod since we changed the default pod template in #7186

**Comment by spadgett** [comment]
/hold cancel

I pushed a fix for the ConsoleExternalLogLink test. @dtaylor113 Another look?

File: frontend/integration-tests/tests/crd-extensions.scenario.ts
Comment (dtaylor113) [inline frontend/integration-tests/tests/crd-extensions.scenario.ts:280]: Is there a BZ to track this future work?

**Review by dtaylor113 — APPROVED** [review]
/lgtm

**Comment by dtaylor113** [comment]
/retest
a11y flake, passing locally.  I'll try to find out why route "/" seems to flake alot.

**Comment by spadgett** [comment]
> a11y flake, passing locally. I'll try to find out why route "/" seems to flake alot.

Protractor tests passed though!

**Comment by dtaylor113** [comment]
Hi @irosenzw, looks like there is an error in the e2e scripts.   
Cypress 'olm' tests aren't supposed to run in the 'e2e' scenerio!  Can you please remove this [line](https://github.com/openshift/console/blob/0cbba0de6fde5f8cecd256563b7badc3b219512a/test-cypress.sh#L43)?  -thanks

**Comment by dtaylor113** [comment]
/retest

Latest CI failure after 4hrs:
```
+ search='Error waiting for Creating Subnetwork: Quota '\''SUBNETWORKS'\'' exceeded'
++ zgrep -E 'Error waiting for Creating Subnetwork: Quota '\''SUBNETWORKS'\'' exceeded' 'installer/.openshift_install.log*'
gzip: installer/.openshift_install.log*.gz: No such file or directory
```
Trying again

**Comment by rhamilto** [comment]
It appears changes in https://github.com/openshift/console/pull/7546 broke the Role Binding CRUD test?  @cyril-ui-developer, your PR was able to merge because the Protractor tests are currently disabled.  Can you verify the [crudBasic test](https://github.com/openshift/console/blob/master/frontend/integration-tests/protractor.conf.ts#L64) runs locally with your changes?

**Comment by dtaylor113** [comment]
/retest
Not sure what the issue is, CI doesn't seem to get to e2e-gcp-console tests:  I see in logs:
```
++ echo -n '==Infrastructure - GCP quota exceeded (route to forum-gcp)=installer/.openshift_install.log*=Error waiting for Creating Subnetwork: Quota '\''SUBNETWORKS'\'' exceeded'
++ cut -f 1 -d =
+ id=
++ echo -n '==Infrastructure - GCP quota exceeded (route to forum-gcp)=installer/.openshift_install.log*=Error waiting for Creating Subnetwork: Quota '\''SUBNETWORKS'\'' exceeded'
++ cut -f 2 -d =
+ covers=
+ [[ -n '' ]]
++ echo -n '==Infrastructure - GCP quota exceeded (route to forum-gcp)=installer/.openshift_install.log*=Error waiting for Creating Subnetwork: Quota '\''SUBNETWORKS'\'' exceeded'
++ cut -f 3 -d =
+ prefix='Infrastructure - GCP quota exceeded (route to forum-gcp)'
++ echo -n '==Infrastructure - GCP quota exceeded (route to forum-gcp)=installer/.openshift_install.log*=Error waiting for Creating Subnetwork: Quota '\''SUBNETWORKS'\'' exceeded'
++ cut -f 4 -d =
+ files='installer/.openshift_install.log*'
++ echo -n '==Infrastructure - GCP quota exceeded (route to forum-gcp)=installer/.openshift_install.log*=Error waiting for Creating Subnetwork: Quota '\''SUBNETWORKS'\'' exceeded'
```
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_console/7446/pull-ci-openshift-console-master-e2e-gcp-console/1339625752278077440/artifacts/e2e-gcp-console/gather-extra/container-logs/test.log

**Comment by dtaylor113** [comment]
/hold
until CI issues are resolved.

**Comment by dtaylor113** [comment]
/hold
investigating possible issue related to the exceeded memory usage

**Comment by dtaylor113** [comment]
cloned this branch locally.
`./test-protractor.sh e2e` ran locally without error.
```
Summary:

Suites:  74 of 74
Specs:   186 of 192 (6 pending)
Expects: 480 (0 failures)
Finished in 594.548 seconds
```

**Comment by dtaylor113** [comment]
/hold
wonder if Cypress OLM failures are causing the timeouts.
hold till #7614 merges

**Comment by dtaylor113** [comment]
Saw a couple of `analyze` failures, now seeing: 
```
level=info msg=Credentials loaded from environment variable "GOOGLE_CLOUD_KEYFILE_JSON", file "/var/run/secrets/ci.openshift.io/cluster-profile/gce.json"
level=fatal msg=failed to fetch Cluster: failed to fetch dependency of "Cluster": failed to generate asset "Platform Provisioning 
Check": metadata.name: Invalid value: "ci-op-lf42bz1h-75d12": record api.ci-op-lf42bz1h-75d12.origin-ci-int-
gce.dev.openshift.com. already exists in DNS Zone (openshift-gce-devel-ci/origin-ci-int-gce-new) and might be in use by another 
cluster, please remove it to continue
error: failed to execute wrapped command: exit status 1
```

**Comment by dtaylor113** [comment]
/hold
Until @irosenzw rebases and force-pushes code to this PR to remove 'ci/prow/kubevirt-plugin' from check/test list.
Also, hold until #7623 has merged which disables OLM tests in `ci/prow/e2e-gcp-console` test suite.

**Comment by dtaylor113** [comment]
Seems like this PR was scaled back to just the fix in test-prow-e2e.sh, unfortunately it seems like the i18n keys need to be adjusted as well?  @irosenzw do you need to run `yarn run i18n` to generate the (old) key?  
```
Error: Failed: Missing i18n key "**Builder Images** are container Images that build source code for a  particular language or framework." in namespace "devconsole" and language "en."
```

**Comment by irosenzw** [comment]
> Seems like this PR was scaled back to just the fix in test-prow-e2e.sh, unfortunately it seems like the i18n keys need to be adjusted as well? @irosenzw do you need to run `yarn run i18n` to generate the (old) key?
> 
> ```
> Error: Failed: Missing i18n key "**Builder Images** are container Images that build source code for a  particular language or framework." in namespace "devconsole" and language "en."
> ```

I ran `yarn run i18n` and push it again. 
please add to this PR whatever else you've added on top of it.

**Comment by spadgett** [comment]
> Seems like this PR was scaled back to just the fix in test-prow-e2e.sh, unfortunately it seems like the i18n keys need to be adjusted as well? @irosenzw do you need to run `yarn run i18n` to generate the (old) key?
> 
> ```
> Error: Failed: Missing i18n key "**Builder Images** are container Images that build source code for a  particular language or framework." in namespace "devconsole" and language "en."
> ```

It looks like my commit that fixed this was somehow lost. I'll push it again.

**Comment by christianvogt** [comment]
@spadgett this PR has been on hold a while because they touched the same files for localization: https://github.com/openshift/console/pull/7529
Do we want to just get that one merged?

**Comment by spadgett** [comment]
> @spadgett this PR has been on hold a while because they touched the same files for localization: #7529
> Do we want to just get that one merged?

@christianvogt I think this one is set again. Do you want to simply remove the "container images" message fix from #7529 so they don't conflict? No need to fix that twice.

We keep regressing on the Protractor tests since they're not running in CI, so I'd really like to prioritize getting this fix in.

/hold cancel

**Comment by dtaylor113** [comment]
/hold cancel
my reasons for holding this PR have been addressed
"Until @irosenzw rebases and force-pushes code to this PR to remove 'ci/prow/kubevirt-plugin' from check/test list.
Also, hold until #7623 has merged which disables OLM tests in ci/prow/e2e-gcp-console test suite."

**Comment by dtaylor113** [comment]
https://github.com/openshift/console/pull/7692 might fix the analyze failure.

**Comment by spadgett** [comment]
>        ✖ does not display the ConsoleExternalLogLink instance on the test pod (3 failures) 

Looks like we timeout the test if the log content itself doesn't load fast enough since `crudView.isLoaded` waits for all loaders to disappear (including the log loader). I think it just timed out in this case. We should look at that as a follow-up. The tests did pass locally for me when I tried them.

https://github.com/openshift/console/blob/master/frontend/integration-tests/tests/crd-extensions.scenario.ts#L308

/retest

**Comment by spadgett** [comment]
@rhamilto I had to add one more small test fix for the project dashboard, can you look again?

**Review by rhamilto — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
/override ci/prow/analyze

analyze job is broken, tracked by https://bugzilla.redhat.com/show_bug.cgi?id=1912115

**Comment by spadgett** [comment]
> Looks like we timeout the test if the log content itself doesn't load fast enough since `crudView.isLoaded` waits for all loaders to disappear (including the log loader). I think it just timed out in this case. We should look at that as a follow-up.

I opened #7709 with a possible fix.

**Pull request merged**
